Short-Answer/Algorithms_Answers.md

```diff
@@ -1 +1,18 @@
-Add your answers to the Algorithms exercises here.
\ No newline at end of file
+Add your answers to the Algorithms exercises here.
+
+Exercise I
+
+a) O(n) as the while loop iterates once from 0 to n^3
```

Comment:
answer is correct but it would have been better. if you could have explained it a bit more.

```python

a)  a = 0                      O(1)
    while (a < n * n * n):     normally O(n^3), but...
      a = a + n * n            here we add n^2 to `a` each time
```

The whole thing is similar to for loop:

```
for (a = 0; a < n * n * n; a += n * n)
```

Those `n`s effectively cancel out to:

```
for (a = 0; a < n; a += 1)     Clearly O(n)


```